src/exodus/wallet.cpp

```diff
@@ -63,25 +63,55 @@ SigmaMintChainState Wallet::GetSigmaMintChainState(const SigmaMintId& id)
     return GetSigmaEntry(id).chainState;
 }
 
+boost::optional<SigmaEntry> Wallet::GetSpendableSigmaMint(uint32_t propertyId, uint8_t denomination)
+{
+    AssertLockHeld(pwalletMain->cs_wallet);
+    std::list<SigmaEntry> allCoins;
+    ListSigmaEntries(propertyId, std::back_inserter(allCoins));
+
+    auto last = std::remove_if(allCoins.begin(), allCoins.end(), [denomination](SigmaEntry const &entry) -> bool {
+        return entry.denomination != denomination
+            || entry.chainState.block < 0
+            || !entry.tx.IsNull();
+    });
+    allCoins.erase(last, allCoins.end());
+
+    if (allCoins.empty()) {
+        return boost::none;
+    }
+
+    auto chosenCoin = std::min_element(allCoins.begin(), allCoins.end(),
+        [](SigmaEntry const &a, SigmaEntry const &b) -> bool {
+            return a.chainState.index < b.chainState.index;
+        }
+    );
+
+    return *chosenCoin;
+}
+
 void Wallet::SetSigmaMintChainState(const SigmaMintId& id, const SigmaMintChainState& state)
 {
     LOCK(pwalletMain->cs_wallet);
 
     auto e = GetSigmaEntry(id);
 
+    if (!e.tx.IsNull()) {
```

Comment:
After discussion we need to remove this check for now and handle it in a separated issue.